lib/openlogi/api/endpoint.rb

```diff
@@ -13,23 +11,22 @@ def resource_class
         raise NotImplementedError
       end
 
-      private
-
       def perform_request(method, resource, options = {})
         Openlogi::Request.new(client, method, resource, options).perform.tap do |response|
-          raise InvalidResponseError.new(response) if response.invalid?
+          client.last_response = response
+          raise BadRequestError.new(response) if response.bad_request?
```

Comment:
We might want to have some slightly more sophisticated error handling, it seems the API returns e.g. a 409 on a create if the created item already exists, etc. but for now I figured this would do.